homeassistant/components/energy/sensor.py

```diff
@@ -416,3 +416,18 @@ def update_config(self, config: dict) -> None:
     def native_unit_of_measurement(self) -> str | None:
         """Return the units of measurement."""
         return self.hass.config.currency
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return the unique ID of the sensor."""
+        entity_registry = er.async_get(self.hass)
+        if (
+            registry_entry := entity_registry.async_get(
+                self._config[self._adapter.entity_energy_key]
+            )
+        ) and registry_entry.unique_id:
+            prefix = registry_entry.unique_id
```

Comment:
This is not unique across integrations, use the UUID (`registry_entry.id`) instead.

Reply:
That would be better indeed :+1: